### PR TITLE
post: ng2 - es6 part 1

### PR DESCRIPTION
**Pull request by cexbrayat**
After the tedious work of porting the asciidoc to amrkdown, here it is:
the first blog post for the Angular 2 serie "The road to Angular 2".

This is the first part of the ES6 chapter, pretty much as is.

I added a few sentences, that I'll submit as a PR to the ebook repo.

### CODE REVIEW

File: _posts/2015-10-06-es6-part-1.md [hunk 1]
```diff
@@ -0,0 +1,400 @@
+---
+layout: post
+title: The road to Angular 2 - ES6 part 1
+author: cexbrayat
+tags: ["Angular 2", "ES6"]
+description: "Angular 2 is not that far now. It has been designed to leverage a lot of new wonderful things in Web development, like ES6. Let's have a look!"
+---
+
+# A gentle introduction to ECMAScript 6
```
Comment (clacote) [inline _posts/2015-10-06-es6-part-1.md]: I find this title useless: the text content is already an introduction.

File: _posts/2015-10-06-es6-part-1.md [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+One of the most emblematic new features, and one that we will vastly use when writing an Angular app: ES6 introduces classes to JavaScript! You can now easily use classes and inheritance in JavaScript. You always could, using prototypal inheritance, but that it was not an easy task, especially for beginners.
+
+Now it's very easy, take a look:
+
+    class Pony {
+      constructor(color) {
+        this.color = color;
+      }
+      toString() {
+        return `${this.color} pony`;
```
Comment (clacote) [inline _posts/2015-10-06-es6-part-1.md]: You never mentioned "templated strings" (with backtick) before.

File: _posts/2015-10-06-es6-part-1.md [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+Static methods can be called only on the class directly:
+
+    let speed = Pony.defaultSpeed();
+
+A class can have getters and setters, if you want to hook on these operations:
+
+    class Pony {
+      get color() {
+        console.log('get color');
+        return this._color;
```
Comment (clacote) [inline _posts/2015-10-06-es6-part-1.md]: This underscore turns every markdown renderer but the GitHub Pages one (Github file view, Atom, ...) crazy: all the following content is in italic. Rendered by Github Pages, it's OK.

**Comment by clacote** [comment]
How are you planning to deal with the free chapter for mailing list member? Do you want to give it for free  first to member (:+1:) or to publish this publicly on our blog first (:cry:)?

File: _posts/2015-10-06-es6-part-1.md (hunk 1 above)
Comment (cexbrayat) [inline _posts/2015-10-06-es6-part-1.md]: Removed!

File: _posts/2015-10-06-es6-part-1.md (hunk 2 above)
Comment (cexbrayat) [inline _posts/2015-10-06-es6-part-1.md]: Right, I added a comment

**Comment by cexbrayat** [comment]
I was thinking about publishing this on the blog and send another chapter to the ML. As your were pointing out, this one is not especially Angular oriented, we can give another one. Is that ok with you?

File: _posts/2015-10-06-es6-part-1.md (hunk 2 above)
Comment (cexbrayat) [inline _posts/2015-10-06-es6-part-1.md]: check https://github.com/Ninja-Squad/ninja-squad.github.com/commit/48a181b063a72ee539b5b75d515f33099e3551cc#diff-7232b255a2e7780d950ac36c217dd9f2R316

**Comment by clacote** [comment]
@cexbrayat: yes, another chapter for mailing list member will do. I just wanted to preserve some kind of "premiere exclusivity" for them

**Pull request merged**